src/libostree/ostree-sysroot-deploy.c

```diff
@@ -1458,20 +1458,30 @@ install_deployment_kernel (OstreeSysroot   *sysroot,
     ostree_bootconfig_parser_set (bootconfig, "linux", boot_relpath);
   }
 
+  val = ostree_bootconfig_parser_get (bootconfig, "options");
+  kargs = _ostree_kernel_args_from_string (val);
+
   if (dest_initramfs_name)
     {
       g_autofree char * boot_relpath = g_strconcat ("/", bootcsumdir, "/", dest_initramfs_name, NULL);
       ostree_bootconfig_parser_set (bootconfig, "initrd", boot_relpath);
     }
-
-  val = ostree_bootconfig_parser_get (bootconfig, "options");
+  else
+    {
+      g_autofree char *ostree_prepare_root = NULL;
+      ostree_prepare_root = g_strdup_printf ("init=/ostree/boot.%d/%s/%s/%d/usr/lib/ostree-boot/ostree-prepare-root /",
```

Comment:
Let's set aside the prepare-root-in-kernel case since I'm having enough trouble keeping the support matrix in my head :smile:  and I doubt anyone is doing it.  The overhead of an additional exec in the startup path is small enough, and avoids having to carry kernel patches, etc.

So, I wouldn't disagree that statically linked users can still use an initramfs, but it'd be odd to have a static switchroot inside an initramfs.  I guess if it's small enough, I could imagine "general purpose" distros like Debian/CentOS always building it statically and allowing people to make initrd-less systems that use ostree, even if it's not going to be the default.  In my world (CentOS/Fedora) though we're not really focused on the embedded cases, so we basically always have an initrd.

Anyways, I'm ok with renaming the build option to `#ifdef BUILDOPT_STATIC_PREPARE_ROOT` which is I guess what all this boils down to :smile: 

So...aside from minor bits, are you thinking we merge this patch as is, or is it more of an RFC and we'll update it with build options later?